src/modules/coach-ai/safety/action-safety.ts

```
@@ -115,6 +119,38 @@ export function assertProposalBatchSafe(
   const destructive = actions.filter((a) => isDestructiveActionType(a.type));
   if (destructive.length === 0) return;
 
+  // Per-item confirmation. Supplying the field at all opts into strict mode:
+  // a client that knows about confirmations must account for every delete in
+  // the batch, otherwise a partial list would be worse than none.
+  const strict =
+    opts.confirmedDeleteIds !== undefined || deleteConfirmationRequired();
+
+  if (strict) {
+    const confirmed = new Set(opts.confirmedDeleteIds ?? []);
+    const unconfirmed = destructive.filter((a) => !a.id || !confirmed.has(a.id));
```

ESLint

[error] Replace `(a)·=>·!a.id·||·!confirmed.has(a.id)` with `⏎······(a)·=>·!a.id·||·!confirmed.has(a.id),⏎····`

(prettier/prettier)